quinn-udp/src/unix.rs

```diff
@@ -425,24 +427,27 @@ fn send(state: &UdpSocketState, io: SockRef<'_>, transmit: &Transmit<'_>) -> io:
         cfg!(apple) || cfg!(target_os = "openbsd") || cfg!(target_os = "netbsd"),
         state.sendmsg_einval(),
     );
-    let n = unsafe { libc::sendmsg(io.as_raw_fd(), &hdr, 0) };
-    if n == -1 {
-        let e = io::Error::last_os_error();
-        match e.kind() {
-            io::ErrorKind::Interrupted => {
-                // Retry the transmission
-            }
-            io::ErrorKind::WouldBlock => return Err(e),
-            _ => {
-                // - EMSGSIZE is expected for MTU probes. Future work might be able to avoid
-                //   these by automatically clamping the MTUD upper bound to the interface MTU.
-                if e.raw_os_error() != Some(libc::EMSGSIZE) {
-                    return Err(e);
+    loop {
+        let n = unsafe { libc::sendmsg(io.as_raw_fd(), &hdr, 0) };
+        if n == -1 {
+            let e = io::Error::last_os_error();
+            match e.kind() {
+                io::ErrorKind::Interrupted => {
+                    // Retry the transmission
+                    continue;
+                }
+                io::ErrorKind::WouldBlock => return Err(e),
+                _ => {
+                    // - EMSGSIZE is expected for MTU probes. Future work might be able to avoid
+                    //   these by automatically clamping the MTUD upper bound to the interface MTU.
+                    if e.raw_os_error() != Some(libc::EMSGSIZE) {
```

Comment:
I know we are doing it this way in the other impls too but this just tripped me up again, thinking that errors are not returned here.

I think this would be clearer if the bool is flipped and we special-case an `Ok(())` return here and default to returning the error.

Let me know if you are up for it, otherwise I can spend some cycles on this next week.

Reply:
This could also be a match-arm guard in which case it would more naturally slot into handling different variants of this error.

(I guess we are using the raw error value here because the equivalent `ErrorKind` is still unstable?)

Reply:
> (I guess we are using the raw error value here because the equivalent `ErrorKind` is still unstable?)

It actually doesn't even seem to be mapped. Sad.

Reply:
I wonder if this case should actually still be special-cased. Should we move this condition further up as well? Sending too large of a packet seems like a worth-while error to report.

Reply:
I don't have a strong opinion, though prefer for any of such change to happen in a separate pull request.